### ISSUE TITLE
Client Generator - Multiple success response codes causes panic

### ISSUE THREAD

**Issue by doswell** [go-swagger/go-swagger#585]
## Problem statement

If a swagger doc includes multiple success responses, the client generated results in a scenario that causes a panic due to attempting to convert betwen invalid generated structs. 

The generated client creates separate structs for types of responses, this causes a conflict when multiple successes are possible, for instance for a AddWidget endpoint that returns either a 200 or 201 response code. 
The client is generated with two structs;
AddWidgetOK
AddWidgetCreated

ReadResponse of the generated responses template will return one of those two objects depending on the response code.
However in the the AddWidget method in the client template, it attempts to return only (The first?) success type, AddWidgetOK. If a 201 created response comes back then a panic occurs. 

## Swagger specification

```
"responses" : {
  "200" : {
    "description" : "(OK): Updated",
    "schema" : {
      "$ref" : "#/definitions/Widget"
    }
  },
  "201" : {
    "description" : "(Created): Added",
    "schema" : {
      "$ref" : "#/definitions/Widget"
    }
  },
  "404" : {
    "description" : "(Not found): Widget not found"
  }
}

```
## Steps to reproduce

Generate a client froma swagger def which has more than one success response code. 
Make a call which returns a success code other than what the client is generated with.

**Comment by doswell** [comment]
I have a working fix at the moment for my needs, however it creates a new *Response struct for returning from the calls. Under that struct are then the *OK or *Created structs.  ie;
```
    type AddGroupResponse struct {
        AddGroupOK *AddGroupOK //200 response
        AddGroupCreated *AddGroupCreated //201 response
        AddGroupNotFound *AddGroupNotFound //404 response
    }
```
The problem with this (in terms of making it a PR as is) is that even for single (success) responses it creates the Response struct, so that would break any existing generated client. 
```
type GetGroupResponse struct {
	GetGroupOK *GetGroupOK //200 response
	GetGroupNotFound *GetGroupNotFound //404 response
}
```

Does the Response struct make sense? is there an alternative? 
   * Should always returning the Response struct instead of a OK/Created be an option? (I personally like this, as any changes to the API/swagger doc may be less impactful) 
   * Should the fields under the struct be prefixed with the method name similar to the models? Or just use the friendly names of the http status codes ie;
```
    type AddGroupResponse struct {
        OK *AddGroupOK //200 response
        Created *AddGroupCreated //201 response
        NotFound *AddGroupNotFound //404 response
    }
```
Since AddGroupOK and AddGroupCreated reall yjust have Payload under them (are there other scnearios where there are more fields?) ie;
```
type AddGroupCreated struct {
	Payload *models.GroupDetails
}
type AddGroupOK struct {
	Payload *models.GroupDetails
}
```
I'm inclined to target a struct of;
```
    type AddGroupResponse struct {
        OK *models.GroupDetails //200 response
        Created *models.GroupDetails //201 response
        NotFound *models.EmptyPayload //404 response
        resultCode int
    }
    func (r *AddGroupResponse) Code() int {
          return r.resultCode
    }
```
?

**Comment by casualjim** [comment]
fixed by: 2b0c1cf25561300f738733bee51948f2a191712a
